2394: Fix Header title for deep linking

SteffiStuffel opened the pull request:
### Short description
removed `t(previousRoute.name)` as a fallback, as we do not have translations for route names, so I think this does never make sense. 
Other option would be to set the back navigation to the city name, what would make sense for opening all deep links, as their back navigation is defined as the home screen of the city, but we would have to load data to convert the citycode to the appropriate city-display name. I think this quite of an overhead, and just having the back icon is fine.

### Resolved issues

<!-- List all issues which should be closed when this PR is merged. -->

Fixes: #2394 

---

<!--
DOR:
- [Release notes](https://github.com/digitalfabrik/integreat-app/blob/main/docs/contributing.md#release-notes) have been added for user visible changes
- Linting: `yarn lint`
- Typescript: `yarn ts:check`
- Prettier: `yarn prettier`
- Unit tests: `yarn test`
- -->

LeandraH reviewed (approved):
Seems like a good enough fix to the problem to me, and not worth the hassle for a pretty rare problem (I assume)

steffenkleinle reviewed (changes requested):
Actually, this breaks quite a few cases:
* news > select one news item (header should be "News" but is empty)
* events > select one events item (header should be "Events" but is empty)
* settings > licenses (header should be "Settings" but is empty)

I agree that the previous way of handling it is maybe not the most obvious/explicit, but this is not an improvement. Perhaps we have to explicitly set the header title in every route.

SteffiStuffel commented:
@steffenkleinle thanks for the input, i fixed it. Also replaced the empty string for categories with the "local information" translation, but can change back to empty if this is preferred.
